client/hosting/hosting-features/components/hosting-features.tsx

```diff
@@ -85,9 +87,9 @@ const HostingFeatures = () => {
 
 	useEffect( () => {
 		if ( isSiteAtomic && ! isPlanExpired ) {
-			page.replace( redirectUrl.current );
+			page.replace( redirectUrl );
 		}
-	}, [ isSiteAtomic, isPlanExpired ] );
+	}, [ isSiteAtomic, isPlanExpired, redirectUrl ] );
```

Comment:
Im actually wondering if this hook is still needed at all? Or if it can be removed entirely.

Reply:
The redirect [was before](https://github.com/Automattic/wp-calypso/pull/93488/files#diff-74b24e4f86658283c3f5964b0ba6a97c57f08c79179ef3af277d4ed425ccdaa9L143-L146) and was used to handle redirect_url from query-params.

I moved it to hook, since it produced warnings in console (since it was in the render flow, outside useEffect) and, as a result, caused unpredictable behaviour for my cases.

It's still necessary for me, I am using it to redirect to "Hosting Overview" and reload the page, as soon as a site becomes atomic.


>And the controller middleware redirects nonSimple sites away from this page on load

I double checked - middleware are not rerun on changes in react-query etc, AFAIU they work separately, only during initial render. Actually, I would remove redirect from the controller and move it to this component, to have redirects in one place, but it's just refactoring and isn't critical.

Reply:
Thanks for clarifying!

> AFAIU they work separately, only during initial render.

indeed.

> It's still necessary for me, I am using it to redirect to "Hosting Overview" and reload the page, as soon as a site becomes atomic.

Thats what I initially assumed, but then when i tested triggering an atomic transfer from this page it went to the separate atomic transfer url/page with the redirect_to already in that URL as a query param, so I was a bit confused if that was supposed to handle the redirect, or the situation in which it doesn't and this page needs to.

Anyways, if this is still necessary this seems fine to leave as is.

> but it's just refactoring and isn't critical.

Agreed.

Reply:
>it went to the separate atomic transfer url/page with the redirect_to already in that URL as a query param

There are two extra cases (steps to use it) described in the description of this PR, take a look when you get a chance, just to know about these edge cases - https://github.com/Automattic/wp-calypso/pull/93488

Reply:
Makes sense, thank you!